### PR TITLE
fix: fix rich text editor loses focus on clicking edit buttons

### PR DESCRIPTION
**Pull request by lightbringer1991**
Ensure that the editor not losing focus when user clicks on styling buttons

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/components/RichTextEditor/ToolbarButton.jsx [hunk 1]
```diff
@@ -8,8 +8,17 @@ const ToolbarButton = ({ onToggle, label, active }) => {
     active,
   });
 
+  const mouseDownHandler = React.useCallback(
+    event => {
+      event.preventDefault();
+      event.stopPropagation();
```
Comment (lightbringer1991) [inline src/components/RichTextEditor/ToolbarButton.jsx:14]: stop event propagation to not trigger `onBlur()` later up the chain

File: src/components/RichTextEditor/index.jsx [hunk 2]
```diff
@@ -40,8 +42,8 @@ const RichTextEditor = ({ className, value, initialValue, onChange, placeholder
   };
 
   return (
-    <div data-testid="rich-text-editor-wrapper" className={classNames} onClick={focusEditor}>
-      <div className="aui--editor-container">
+    <div data-testid="rich-text-editor-wrapper" className={classNames}>
+      <div className="aui--editor-container" onClick={focusEditor}>
```
Comment (lightbringer1991) [inline src/components/RichTextEditor/index.jsx:46]: we just want to focus when clicking on the textbox itself, not the whole component.

**Review by Harris-YWang — APPROVED** [review]
LGTM

**Review by pphminions — APPROVED** [review]
LGTM 🍏

**Review by superpower1 — APPROVED** [review]

**Comment by pphminions** [comment]
merging 🚀

**Pull request merged**